com.ibm.streamsx.metrics/impl/java/src/com/ibm/streamsx/metrics/MemoryMetricsSource.java

```diff
@@ -0,0 +1,535 @@
+import com.ibm.streams.management.job.PeMXBean;
+import com.ibm.streams.operator.AbstractOperator;
+import com.ibm.streams.operator.OperatorContext;
+import com.ibm.streams.operator.OutputTuple;
+import com.ibm.streams.operator.StreamingData.Punctuation;
+import com.ibm.streams.operator.StreamingOutput;
+import com.ibm.streams.operator.model.Libraries;
+import com.ibm.streams.operator.model.OutputPortSet;
+import com.ibm.streams.operator.model.OutputPortSet.WindowPunctuationOutputMode;
+import com.ibm.streams.operator.model.OutputPorts;
+import com.ibm.streams.operator.model.Parameter;
+import com.ibm.streams.operator.model.PrimitiveOperator;
+import com.ibm.streamsx.metrics.internal.OperatorConfiguration;
+import com.ibm.streamsx.metrics.internal.TupleContainer;
+
+/**
+ * A source operator that does not receive any input streams and produces new tuples. 
+ * The method <code>produceTuples</code> is called to begin submitting tuples.
+ * <P>
+ * For a source operator, the following event methods from the Operator interface can be called:
+ * </p>
+ * <ul>
+ * <li><code>initialize()</code> to perform operator initialization</li>
+ * <li>allPortsReady() notification indicates the operator's ports are ready to process and submit tuples</li> 
+ * <li>shutdown() to shutdown the operator. A shutdown request may occur at any time, 
+ * such as a request to stop a PE or cancel a job. 
+ * Thus the shutdown() may occur while the operator is processing tuples, punctuation marks, 
+ * or even during port ready notification.</li>
+ * </ul>
+ * <p>With the exception of operator initialization, all the other events may occur concurrently with each other, 
+ * which lead to these methods being called concurrently by different threads.</p> 
+ */
+@PrimitiveOperator(
+		name="MemoryMetricsSource",
+		namespace="com.ibm.streamsx.metrics",
+		description=MemoryMetricsSource.DESC_OPERATOR
+		)
+@OutputPorts({
+	@OutputPortSet(
+			cardinality=1,
+			optional=false,
+			windowPunctuationOutputMode=WindowPunctuationOutputMode.Generating,
+			description=MemoryMetricsSource.DESC_OUTPUT_PORT
+			)
+})
+@Libraries({
+	// As described here:
+	// http://www.ibm.com/support/knowledgecenter/en/SSCRJU_4.2.0/com.ibm.streams.dev.doc/doc/jmxapi-start.html
+	// Environment variables (@...@) are evaluated during compile-time and must
+	// be identical in the run-time environment.
+	"@STREAMS_INSTALL@/lib/com.ibm.streams.management.jmxmp.jar",
+	"@STREAMS_INSTALL@/lib/com.ibm.streams.management.mx.jar",
+	"@STREAMS_INSTALL@/ext/lib/jmxremote_optional.jar"
+	})
+public class MemoryMetricsSource extends AbstractOperator {
+
+	// ------------------------------------------------------------------------
+	// Documentation.
+	// Attention: To add a newline, use \\n instead of \n.
+	// ------------------------------------------------------------------------
+	
+	static final String DESC_OPERATOR = 
+			"The MemoryMetricsSource operator uses the "
```

Comment:
The intention of the MetricsSource operator was to provide all kind of metrics, also the PE metrics. Therefore, please enhance the MetricsSource operator with the PE metrics instead of creating a new operator.